code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -80,9 +80,12 @@
 			user << "<span class='danger'>[src] buzzes and beeps.</span>"
 
 /mob/living/simple_animal/bot/cleanbot/process_scan(obj/effect/decal/cleanable/D)
+	. = list()
 	for(var/T in target_types)
 		if(istype(D, T))
-			return D
+			. += D
+	if(.)

```

Comment:
Loops and acting once have been proven to be faster than other approaches, so it certainly seems reasonable on the face of things.

but it does look hideous and would probably be best avoided by using a real list var for the entire proc, and then ```. = that_List.len ? pick(that_list) : null``` at the end